### PR TITLE
21536 - Passing short Name Id parameter to find refunds

### PR DESCRIPTION
**Pull request by rodrigo-barraza**
*Issue #:*
https://github.com/bcgov/entity/issues/21536

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: pay-api/src/pay_api/services/eft_refund.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def create_shortname_refund(request: dict, **kwargs):
     @staticmethod
     def get_shortname_refunds(data: EFTShortNameRefundGetRequest):
         """Get all refunds."""
-        refunds = EFTRefundModel.find_refunds(data.statuses)
+        refunds = EFTRefundModel.find_refunds(data.statuses, data.short_name_id)
```
Comment (ochiu) [inline pay-api/src/pay_api/services/eft_refund.py:83]: Update tests for short_name_id search please

File: pay-api/tests/unit/api/test_eft_short_names.py [hunk 2]
```diff
@@ -1071,12 +1071,12 @@ def test_post_shortname_refund_invalid_request(client, mocker, jwt):
...
         ("", "get_all", 3),
         (
-            f"?status={EFTShortnameRefundStatus.APPROVED.value},{EFTShortnameRefundStatus.PENDING_APPROVAL.value}",
+            f"?statuses={EFTShortnameRefundStatus.APPROVED.value},{EFTShortnameRefundStatus.PENDING_APPROVAL.value}",
             "status_filter_multiple",
             2,
         ),
         (
-            f"?status={EFTShortnameRefundStatus.DECLINED.value}",
+            f"?statuses={EFTShortnameRefundStatus.DECLINED.value}",
```
Comment (ochiu) [inline pay-api/tests/unit/api/test_eft_short_names.py:1079]: additional parameters for short_name_id search?

File: pay-api/src/pay_api/dtos/eft_shortname.py [hunk 3]
```diff
@@ -71,10 +71,14 @@ class EFTShortNameRefundGetRequest:
...
     statuses: List[str]
+    short_name_id: int = None
 
     @classmethod
     def from_dict(cls, data: dict):
         """Convert from request json to EFTShortNameRefundDTO."""
-        input_string = data.get("status", "")
+        input_string = data.get("statuses", "")
+        short_name_id = None
+        if data.get("shortNameId", None) is not None:
```
Comment (seeker25) [inline pay-api/src/pay_api/dtos/eft_shortname.py:81]: please use serialization instead of doing this

**Review by seeker25 — CHANGES_REQUESTED** [review]
comment above

File: pay-api/src/pay_api/dtos/eft_shortname.py (hunk 3 above)
Comment (seeker25) [reply pay-api/src/pay_api/dtos/eft_shortname.py:81]: we eventually want to get rid of dto.state = dto.state.split(",") etc somehow in the converter

File: pay-api/tests/unit/api/test_eft_short_names.py [hunk 4]
```diff
@@ -1071,12 +1071,12 @@ def test_post_shortname_refund_invalid_request(client, mocker, jwt):
     [
         ("", "get_all", 3),
         (
-            f"?status={EFTShortnameRefundStatus.APPROVED.value},{EFTShortnameRefundStatus.PENDING_APPROVAL.value}",
+            f"?statuses={EFTShortnameRefundStatus.APPROVED.value},{EFTShortnameRefundStatus.PENDING_APPROVAL.value}",
```
Comment (seeker25) [inline pay-api/tests/unit/api/test_eft_short_names.py:1074]: good change I think, status is singular and you'd think you'd repeat the param

File: pay-api/src/pay_api/dtos/eft_shortname.py (hunk 3 above)
Comment (seeker25) [reply pay-api/src/pay_api/dtos/eft_shortname.py:81]: I'll see if we can put in:

typical register structure unstructure hook - new type DelimitedString

**Review by seeker25 — APPROVED** [review]
perfect thank you

**Pull request merged**